client/my-sites/domains/domain-management/list/domain-row.jsx

```diff
@@ -387,9 +391,16 @@ class DomainRow extends PureComponent {
 							? translate( 'View transfer' )
 							: translate( 'View settings' ) }
 					</PopoverMenuItem>
-					<PopoverMenuItem icon="info-outline" onClick={ this.goToDNSManagement }>
-						{ translate( 'Manage DNS' ) }
-					</PopoverMenuItem>
+					{ ! (
+						domain.type === domainTypes.SITE_REDIRECT ||
+						domain.transferStatus === transferStatus.PENDING_ASYNC
+					) &&
+						domain.canManageDnsRecords &&
+						domain.transferStatus !== transferStatus.PENDING_ASYNC && (
```

Comment:
```suggestion
					{ ! (
						domain.type === domainTypes.SITE_REDIRECT ||
						domain.transferStatus === transferStatus.PENDING_ASYNC
					) &&
						domain.canManageDnsRecords && (
```

This can be simplified, because if `domain.transferStatus === transferStatus.PENDING_ASYNC` is true the first part of the condition will be false

Reply:
These conditions are starting to get unwieldy 😄 I think the "right" way would be to check just `domain.canManageDnsRecords` here and verify all conditions in the backend, but that's for another time. This works fine as is 🙂

Reply:
Oh yeah, these conditions should be the same as the ones in the `renderDnsRecords` method in `domain-management/settings/index.tsx`, so I think we need to update there as well